includes/class-payment-information.php

```diff
@@ -235,7 +236,13 @@ public static function get_payment_method_from_request( array $request ): string
 	 * @return \WC_Payment_Token|NULL
 	 */
 	public static function get_token_from_request( array $request ) {
-		$token_request_key = 'wc-' . \WC_Payment_Gateway_WCPay::GATEWAY_ID . '-payment-token';
+		if ( ! empty( $request['payment_method'] ) && Sepa::GATEWAY_ID === $request['payment_method'] ) {
+			$request_gateway_id = Sepa::GATEWAY_ID;
+		} else {
+			$request_gateway_id = \WC_Payment_Gateway_WCPay::GATEWAY_ID;
```

Comment:
What do you think about following this approach?

```php
		$payment_method = $request['payment_method'] ?? null;
		if ( Sepa::GATEWAY_ID === $payment_method ) {
			$request_gateway_id = Sepa::GATEWAY_ID;
		} else {
			$request_gateway_id = Card::GATEWAY_ID;
		}
```

It's essentially the same, but with less cognitive load.